lib/abci/handlers/endBlockHandlerFactory.js

```diff
@@ -0,0 +1,37 @@
+const {
+  abci: {
+    ResponseEndBlock,
+  },
+} = require('abci/types');
+
+/**
+ * End block ABCI handler
+ *
+ * @param {BlockchainState} blockchainState
+ * @param {BaseLogger} logger
+ *
+ * @return {endBlockHandler}
+ */
+function endBlockHandlerFactory(
+  blockchainState,
+  logger,
+) {
+  /**
+   * @typedef endBlockHandler
+   *
+   * @param {abci.RequestEndBlock} request
+   * @return {Promise<abci.ResponseEndBlock>}
+   */
+  async function endBlockHandler({ latestCoreChainLock }) {
```

Comment:
We need a singleton class LatestCoreChainLock to store ChainLock primitive that should be passed here as a dependency (as an endBlockHandlerFactory argument)